### PR TITLE
Lido mainnet

### PR DESCRIPTION
**Pull request by miklt**
Fixes #OE-2

I'm including the Lido Subgraph on mainnet.
The Lido subgraph with just the Lido entities, inside the megagraph structure, is currently deployed at:
 https://thegraph.com/hosted-service/subgraph/miklt/megagraph-lido-mainnet 

The full megagraph, with the Lido Subgraph is deployed at: 
https://thegraph.com/hosted-service/subgraph/miklt/megagraph?version=pending

### CODE REVIEW

File: src/Lido/mappings/Token.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  // new lido event.
+
+  let entity = LidoTokenData.load(event.transaction.hash.toHex());
+  if (entity == null) {
+    entity = new LidoTokenData(event.transaction.hash.toHex());
+    entity.blockNumber = event.block.number;
+    entity.blockTimestamp = event.block.timestamp;
+  }
+
+  let fromZeros = event.params.from == Address.fromString("0x0000000000000000000000000000000000000000");
```
Comment (uwe) [inline src/Lido/mappings/Token.ts:16]: ZERO_ADDRESS

File: src/Lido/mappings/lidoConstants.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { LidoToken } from "../../../generated/LidoTokenstETH/LidoToken";
+import { LidoOracle } from "../../../generated/LidoOracle/LidoOracle";
+import { LidoNodeOperatorsRegistry } from "../../../generated/LidoOracle/LidoNodeOperatorsRegistry";
+export { LidoOracleAddress, LidoTreasuryAddress, LidoTokenAddress, LidoNodeOperatorsRegistryAddress };
+export const loadLidoContract = (): LidoToken => LidoToken.bind(LidoTokenAddress);
+export const loadOracleContract = (): LidoOracle => LidoOracle.bind(LidoOracleAddress);
+export const loadNORegistryContract = (): LidoNodeOperatorsRegistry =>
+  LidoNodeOperatorsRegistry.bind(LidoNodeOperatorsRegistryAddress);
+
+// 1 ETH in WEI
```
Comment (uwe) [inline src/Lido/mappings/lidoConstants.ts:18]: I don't like that file. You have some "constants" here and some in `src/utils/constants.ts`.
And it is a long way of specifying 32 ETH. Just use the end result and add a comment "32 ETH".

File: src/Lido/mappings/Token.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  entity.save();
+}
+export function handleSubmitted(event: Submitted): void {
+  let totals = LidoTotals.load("");
+
+  let isFirstSubmission = !totals;
+
+  if (!totals) {
```
Comment (uwe) [inline src/Lido/mappings/Token.ts:54]: If you declare a variable for this, then also use it.

File: src/Lido/mappings/Token.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import { Address } from "@graphprotocol/graph-ts";
+import { Transfer, Submitted, Withdrawal } from "../../../generated/LidoTokenstETH/LidoToken";
+import { LidoTokenData, LidoRewardData, LidoTotals } from "../../../generated/schema";
+import { LIDO_DUST_BOUNDARY } from "./lidoConstants";
+import { ZERO_BI, LidoTreasuryAddress } from "../../utils/constants";
+export function handleTransfer(event: Transfer): void {
+  // new lido event.
```
[the remark this answers is not in the document]
Comment (miklt) [reply src/Lido/mappings/Token.ts:7]: I forgot to remove that comment.

File: src/Lido/mappings/Token.ts (hunk 3 above)
Comment (miklt) [reply src/Lido/mappings/Token.ts:54]: I refactored the file in a new commit.

**Review by uwe — APPROVED** [review]

**Pull request merged**